bidiff-squashfs/cli/src/main.rs

```diff
@@ -116,3 +157,54 @@ fn run_patch(args: PatchCommand) -> Result<()> {
 
     Ok(())
 }
+
+async fn run_mk_ota(args: OtaCommand) -> Result<()> {
+    if args.out.exists() {
+        let is_empty = is_empty_dir(&args.out).await.wrap_err_with(|| {
+            format!("out dir {} cannot be read", args.out.display())
+        })?;
+        ensure!(is_empty, "out dir {} must be empty", args.out.display());
+    } else {
+        tokio::fs::create_dir_all(&args.out)
+            .await
+            .wrap_err_with(|| {
+                format!("failed to create out dir `{}`", args.out.display())
+            })?;
+    }
+
+    let tmpdir = args.download_dir.is_none().then(|| {
+        tempfile::tempdir_in(".")
```

Comment:
Isn't this a blocking call?

Reply:
yep, it is. I figured no one would care so I did it anyway :sweat_smile: